lib/auth/auth.go

```diff
@@ -3398,6 +3412,25 @@ func (a *Server) DeleteKubernetesCluster(ctx context.Context, name string) error
 	return nil
 }
 
+// SubmitUsageEvent submits an external usage event.
+func (a *Server) SubmitUsageEvent(ctx context.Context, req *proto.SubmitUsageEventRequest) error {
+	username, err := GetClientUsername(ctx)
+	if err != nil {
+		return trace.Wrap(err)
```

Comment:
If we can't get the username, is the event useless to us? Or would we prefer to emit with an empty username?

Reply:
I'm not sure what situations there will ever be an empty username, even non-human users (like the proxy) have something even if it's nonsense. I'm inclined to fail in that case so there's at least some visibility since it's probably a result of some other problem.